### PR TITLE
feat: 🎸 Credential type selector refactor

### PR DESCRIPTION
**Pull request by dvonanderson**
✅ Closes: ICU-6839

:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/ICU-6839)

## Description
When editing a credential or viewing the details screen, the credential type should be displayed as a readonly text field

:technologist: [Admin preview](https://www.figma.com/file/BB6W9sZwp75aOsXgCN8ecb/JSON-Credentials?node-id=1401%3A13584)

#### Not the exact image of a credential type but basically the same look.  This is the previous look of how the `type` was displayed on the `detailed-view` screen and the `edit` screen.  The card was just disabled and non-functional
Screen Shot 2022-11-14 at 12 52 09

#### Now we have a `disabled` `readonly` text field that displays the label "Type" and what type of credential it is on `detailed-view` and `edit` screens.
Screen Shot 2022-11-14 at 12 55 30

### CODE REVIEW

**Review by lisbet-alvarez — APPROVED** [review]
LGTM 👍

File: ui/admin/app/components/form/credential/ssh_private_key/index.hbs [hunk 1]
```diff
@@ -66,24 +66,17 @@
...
-    </:body>
-  </form.fieldset>
+      </:body>
+    </form.fieldset>
+  {{else}}
+    <form.input
+      @name='type'
+      @label={{t 'form.type.label'}}
+      @value={{t (concat 'resources.credential.types.' @model.type)}}
+      @disabled={{true}}
```
Comment (randallmorey) [inline ui/admin/app/components/form/credential/ssh_private_key/index.hbs:76]: Is disabled necessary if it's readonly?

Comment (dvonanderson) [reply ui/admin/app/components/form/credential/ssh_private_key/index.hbs:76]: @randallmorey not it is not.  thanks for the catch

**Review by cameronperera — APPROVED** [review]
LGTM

**Review by calcaide — APPROVED** [review]
LGTM!

**Comment by randallmorey** [comment]
👋  Great work!  I know we are working on an improvement on this design and it was uncertain if we should wait or not.  Ultimately this is a safe improvement to merge in its current form.  Thank you!

**Pull request merged**